moveit_planners/trajopt/test/trajectory_test.cpp

```diff
@@ -0,0 +1,70 @@
+/*********************************************************************
+ * Software License Agreement
+ *
+ *  Copyright (c) 2019, PickNik Consulting
+ *  All rights reserved.
+ *
+ *********************************************************************/
+
+/* Author: Omid Heidari, Dave Coleman
+   Desc:   Test the trajectory for various numbers of timesteps
+ */
+
+// C++
+#include <iostream>
+#include <string>
+
+// ROS
+#include <ros/ros.h>
+
+// Testing
+#include <gtest/gtest.h>
+
+#include <trajopt/common.hpp>
+
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/robot_state/robot_state.h>
+
+#include "trajopt_interface/problem_description.h"
+
+class TrajectoryTest : public ::testing::Test
+{
+public:
+  TrajectoryTest()
+  {
+  }
+
+protected:
+protected:
+  robot_model::RobotModelPtr robot_model_;
+
+};  // class TrajectoryTest
+
+TEST_F(TrajectoryTest, GenerateInitialTrajectoryDimensions)
+{
+  planning_scene::PlanningSceneConstPtr planning_scene;  // not being used. no initialization needed.
+  trajopt_interface::ProblemInfo pci(planning_scene, "panda_arm");
+  pci.basic_info.n_steps = 30;
+  std::vector<double> joint_values = { 0.4, 0.3, 0.5, -0.55, 0.88, 1.0, -0.075 };
+  trajopt::TrajArray init_traj;
+
+  generateInitialTrajectory(pci, joint_values, init_traj);
+
+  EXPECT_EQ(init_traj.cols(), joint_values.size());
+  EXPECT_EQ(init_traj.rows(), pci.basic_info.n_steps);
+}
```

Comment:
please add significantly more tests

Reply:
I will do this when we want to merge the next PR with no tesseract dependency, just because I need more time for creating tests

Reply:
This is not an acceptable reason, unit tests are EXTREMELY easy to skip and never come back to (look at all of moveit).

They are easiest to write when the code is fresh in your head / not yet merged. Please add tests before merging this.

Reply:
added a unit test for goal tolerance